frontend/src/pages/manage-roadmap/CurrentRoadmap.jsx

```diff
@@ -69,6 +81,80 @@ const CurrentRoadmap = ({ setCurrentTab, setEditDatapoint }) => {
       });
   };
 
+  const handleDownloadButton = ({ id, form }) => {
+    setDownloadLoading(id);
+    api
+      .get(`/roadmap-download/${id}`)
+      .then((res) => {
+        setDownloadData(res?.data);
+        setTimeout(() => {
+          const print = document.getElementById("print-iframe");
+          if (print) {
+            const today = moment().format("MMMM Do YYYY");
+            const title = `${form}_${today}`;
+            // for firefox
+            print.contentDocument.title = title;
+            // hack for chrome
+            document.title = title;
+            print.focus();
+            print.contentWindow.print();
+          }
+        }, 2500);
+      })
+      .catch((e) => {
+        const { status, statusText } = e.response;
+        console.error(status, statusText);
+        if (status === 410) {
+          notification.warning({
+            message: "Expired",
+            description: "Download link has been expired.",
+          });
+        } else {
+          notify({
+            type: "error",
+            message: "Something went wrong.",
+          });
+        }
+      })
+      .finally(() => {
+        setTimeout(() => {}, 2500);
+        setDownloadLoading(null);
+      });
+  };
+
+  if (window.frames?.["print-iframe"]) {
+    window.frames["print-iframe"].contentWindow.onafterprint = function () {
+      document.title = "ISCO";
+      setDownloadData(null);
+    };
+    setTimeout(() => {
+      document.title = "ISCO";
+    }, 1000);
+  }
+
+  const handleLoad = (event) => {
```

Comment:
Hi @navins94 , 
We could turn this function into a util function since we use this on both download data & download roadmap data.